nrfutil/modules.ts

```diff
@@ -0,0 +1,166 @@
+                        );
+                }
+
+                moduleSandbox.onLogging((evt, pid) => {
+                    const logger = getNrfutilLogger();
+                    const formatMsg = (msg: string) =>
+                        `${
+                            pid && moduleSandbox?.logLevel === 'trace'
+                                ? `[PID:${pid}] `
+                                : ''
+                        }${msg}`;
+
+                    switch (evt.level) {
+                        case 'TRACE':
+                            logger?.verbose(formatMsg(evt.message));
+                            break;
+                        case 'DEBUG':
+                            logger?.debug(formatMsg(evt.message));
+                            break;
+                        case 'INFO':
+                            logger?.info(formatMsg(evt.message));
+                            break;
+                        case 'WARN':
+                            logger?.warn(formatMsg(evt.message));
+                            break;
+                        case 'ERROR':
+                            logger?.error(formatMsg(evt.message));
+                            break;
+                        case 'CRITICAL':
+                            logger?.error(formatMsg(evt.message));
+                            break;
+                        case 'OFF':
+                        default:
+                            // Unreachable
+                            break;
+                    }
+                });
+
+                const initialLogLevel = getIsLoggingVerbose()
+                    ? 'trace'
+                    : fallbackLevel;
+                moduleSandbox.setLogLevel(initialLogLevel);
+            }
+
+            const box = await promiseModuleSandbox;
+
+            return box;
+        },
+    };
+};
+
+const modules: Record<
+    string,
+    {
+        isInitialised: () => boolean;
+        get: () => Promise<NrfutilSandbox>;
+    }
+> = {};
+if (process.env.NODE_ENV !== 'test' && !isLauncher()) {
+    const nrfutil = packageJsonApp().nrfConnectForDesktop.nrfutil;
+    if (nrfutil) {
+        Object.keys(nrfutil).forEach(module => {
+            modules[module] = getModuleSandbox(module);
+        });
+    }
+}
+
+export const getModule = (module: string) => {
+    if (modules[module]) {
+        return modules[module].get();
+    }
+
+    throw new Error(`Module ${module} not found`);
+};
+
+export const isModuleInitialised = (module: string) =>
+    !!modules[module]?.isInitialised();
+
+const getAllIninitialisedModules = async () => {
+    const moduleSandboxes: NrfutilSandbox[] = [];
+    await Object.values(modules).reduce(
+        (acc, val) =>
+            acc.then(async () => {
+                if (val.isInitialised()) {
+                    moduleSandboxes.push(await val.get());
+                }
+            }),
+        Promise.resolve()
+    );
+
+    return moduleSandboxes;
+};
```

Comment:
From what I understand the `reduce` serialises the promises. But what is the advantage of this? Creating them in parallel seems simpler and potentially faster:
```ts
const getAllIninitialisedModules = () =>
    Promise.all(
        Object.values(modules)
            .filter(module => module.isInitialised)
            .map(module => module.get())
    );
```